### PR TITLE
feat: Added links filter

### LINKED ISSUE #447: Feat(Moderation): Add link spam filter

### PR DESCRIPTION
**Pull request by kyranet**
I agree I might have gone a little overkill.

Fixes #447

### CODE REVIEW

**Comment by favna** [comment]
I agree you've gone way overkill lmao. But as long as it works well I see no problem there.

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,9 @@ node_modules/
 dist/
 bwd/
 
+# Ignore JavaScript files
+src/**/*.js
```
Comment (favna) [inline .gitignore]:
```suggestion
**/*.js
```

Maybe just ignore all JS files repo-wide?

File: src/commands/Management/Message Filters/linkMode.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	protected keyEnabled = GuildSettings.Selfmod.Links.Enabled;
+	protected keySoftAction = GuildSettings.Selfmod.Links.SoftAction;
+	protected keyHardAction = GuildSettings.Selfmod.Links.HardAction;
+	protected keyHardActionDuration = GuildSettings.Selfmod.Links.HardActionDuration;
+	protected keyThresholdMaximum = GuildSettings.Selfmod.Links.ThresholdMaximum;
+	protected keyThresholdDuration = GuildSettings.Selfmod.Links.ThresholdDuration;
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['link-mode', 'lmode'],
```
Comment (favna) [inline src/commands/Management/Message Filters/linkMode.ts]:
```suggestion
			aliases: ['link-mode', 'lmode', 'linkfilter', 'extlinks', 'externallinks'],
```

This way the aliases cover the same-functioning command from Ribbon

File: src/languages/es-ES.ts [hunk 3]
```diff
@@ -1022,6 +1022,32 @@ export default class extends Language {
...
+				'enable',
+				'disable',
+				'action alert',
+				'punishment ban',
+				'punishment mute',
+				'punishment-duration 1m',
+				'threshold-maximum 5',
+				'threshold-duration 30s'
+			]
+		}),
```
Comment (favna) [inline src/languages/es-ES.ts:1050]: These should be translated

File: src/languages/es-ES.ts [hunk 4]
```diff
@@ -146,10 +146,10 @@ export default class extends Language {
...
 		RESOLVER_INVALID_USER: name => `${name} debe ser una mención o una id de usuario válida.`,
 		RESOLVER_STRING_SUFFIX: ' carácteres',
-		RESOLVER_MINMAX_EXACTLY: (name, min, suffix) => `${name} debe ser exactamente de ${min}${suffix}.`,
-		RESOLVER_MINMAX_BOTH: (name, min, max, suffix) => `${name} debe ser de exactamente entr ${min} y ${max}${suffix}.`,
-		RESOLVER_MINMAX_MIN: (name, min, suffix) => `${name} debe ser mayor que ${min}${suffix}.`,
-		RESOLVER_MINMAX_MAX: (name, max, suffix) => `${name} debe ser menor que ${max}${suffix}.`,
+		RESOLVER_MINMAX_EXACTLY: (name, min) => `${name} must be exactly ${min}.`,
+		RESOLVER_MINMAX_BOTH: (name, min, max, inclusive) => inclusive ? `${name} must be between ${min} and ${max} inclusively.` : `${name} must be between ${min} and ${max} exclusively.`,
+		RESOLVER_MINMAX_MIN: (name, min, inclusive) => inclusive ? `${name} must be greater than ${min} inclusively.` : `${name} must be greater than ${min} exclusively.`,
+		RESOLVER_MINMAX_MAX: (name, max, inclusive) => inclusive ? `${name} must be less than ${max} inclusively` : `${name} must be less than ${max} exclusively.`,
```
Comment (favna) [inline src/languages/es-ES.ts:152]: These should be translated. Especially because they used to be.

File: tsconfig.json [hunk 5]
```diff
@@ -19,9 +19,6 @@
 	],
 	"exclude": [
 		"**/node_modules",
-		"**/build",
-		"**/bin",
-		"**/out",
```
Comment (favna) [inline tsconfig.json:24]: Not that I really care but why were these removed?

Comment (kyranet) [reply tsconfig.json:24]: Those are not used anywhere, we only use `dist`, so no reason for those to exist.
